codeworld-api/src/CodeWorld/Picture.hs

```diff
@@ -57,6 +57,46 @@ data Picture
            !Double
            !Bool
            !Bool
+    | Polygon CallStack
+           [Point]
+           !Double
```

Comment:
Removing these extra parameters is part of the point of splitting this up.  You don't need to remember the width of a polygon, nor whether it's closed or smooth, because a polygon by definition has a width of 0, is closed, and is not smooth.  You should just need the `CallStack`, list of points, and ONLY for the thick variants, the thickness.

Reply:
silly question, but for ```Driver.hs```, do I need the n and w parameters (in ```pictureToDrawing```, etc)? They're both Doubles. I remember you had said something last week about having values (eg 1 through 4) specific to each type of Path value.

Reply:
If I were you, I'd focus on removing them from the `Picture` type.  In `pictureToDrawing`, you can infer them from the `Picture` constructor.  For instance, if the constructor is `Curve`, then width is 0, smooth is true, and closed is false

Reply:
I see. Thanks. I'll work on it this evening.

Reply:
I've removed them but am still having problems resolving the type (specifically ```n``` and ```w```). I could push up what I have if that helps. I had tried to previously replace them with literals (1 through 4), but it didn't type-check. I understand that the n and w are Doubles and I've removed them from the ```Picture``` input types in ```picToObj'``` and ```pictureToDrawing```. So currently, I have something like ```pictureToDrawing (SolidPolygon _ pts) = Shape $ pathDrawer pts w True False``` and my w is not in scope because I deleted it from ```Picture.hs```. But it still needs a double. Similarly with the input parameters of say 
````Polygon cs pts -> do 
   obs <- init "polygon"
ptsJS <- pointsToArr pts 
setProps
[("points",ptsJS)
, ("width", pToJSVal w)
, ("closed", pToJSVal True)
, ("smooth", pToJSVal False)
]
obj
retVal obj
```
This is where I'm stuck. I'm trying to resolve the n/w Bool types for these, considering I've deleted them from the ```Picture``` arguments. Don't know if this makes sense.

Reply:
I don't know what you mean by "the `n`".  The w parameter is still needed for thick shapes, but it is always zero (so not needed) for the others.

Please do push your current code when you have a question.  It helps a lot.

Reply:
I see now.  It looks like `n` is used instead of `w` sometimes, but they mean the same thing.

Reply:
Just pushed it with the heading "this does not typecheck". I believe that ```Picture.hs``` is OK. I'm working on ```Driver.hs``` at the moment